core/core-backend/src/main/java/io/dataease/commons/utils/SqlparserUtils.java

```diff
@@ -606,7 +606,11 @@ private static String transFilter(SqlVariableDetails sqlVariableDetails, Map<Lon
                     && sqlVariableDetails.getDeType() == 0) {
                 return "N'" + String.join("', N'", sqlVariableDetails.getValue()) + "'";
             } else {
-                return "'" + String.join("','", sqlVariableDetails.getValue()) + "'";
+                if (sqlVariableDetails.getDeType() == 2 || sqlVariableDetails.getDeType() == 3) {
+                    return String.join(",", sqlVariableDetails.getValue());
+                } else {
+                    return "'" + String.join("','", sqlVariableDetails.getValue()) + "'";
+                }
             }
         } else if (sqlVariableDetails.getOperator().equals("between")) {
             if (sqlVariableDetails.getDeType() == 1) {
```

Comment:
Code has potential for improvement:

1. Use single quotes consistently instead of double quotes.
2. For cases where the value consists of multiple strings separated by commas or spaces (`"abc", \"def"`, `"one two three"`), use `String.join(",", sqlVariableDetails.getValue());` to avoid unnecessary conversion between string and integer types when constructing SQL queries.

To optimize this code further, please consult with a professional programmer who is familiar with your database structure and needs. It's important to ensure that all variable names match correctly within your data schema in order to avoid any future errors during execution.